src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -104,6 +124,67 @@ public static ImmutableArray<ILocalSymbol> GetDeclaredVariables(this IVariableDe
 
             return arrayBuilder.ToImmutableAndFree();
         }
+
+        /// <summary>
+        /// Get an optional argument name for a named argument to the given <paramref name="dynamicExpression"/> at the given <paramref name="index"/>.
+        /// </summary>
+        /// <param name="dynamicExpression">Dynamic or late bound expression.</param>
+        /// <param name="index">Argument index.</param>
+        public static string GetArgumentName(this IHasDynamicArgumentsExpression dynamicExpression, int index)
+        {
+            if (dynamicExpression == null)
+            {
+                throw new ArgumentNullException(nameof(dynamicExpression));
+            }
+
+            var expression = (HasDynamicArgumentsExpression)dynamicExpression;
+            if (expression.ArgumentNames.IsDefaultOrEmpty)
+            {
+                return null;
+            }
+
+            if (index < 0 || index >= expression.ArgumentNames.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+
+            return expression.ArgumentNames[index];
+        }
+
+        /// <summary>
+        /// Get an optional argument <see cref="RefKind"/> for an argument at the given <paramref name="index"/> to the given <paramref name="dynamicExpression"/>.
+        /// Returns a non-null argument <see cref="RefKind"/> for C#.
+        /// Always returns null for VB as <see cref="RefKind"/> cannot be specified for an the argument in VB.
+        /// </summary>
+        /// <param name="dynamicExpression">Dynamic or late bound expression.</param>
+        /// <param name="index">Argument index.</param>
+        public static RefKind? GetArgumentRefKind(this IHasDynamicArgumentsExpression dynamicExpression, int index)
+        {
+            if (dynamicExpression == null)
+            {
+                throw new ArgumentNullException(nameof(dynamicExpression));
+            }
+
+            var expression = (HasDynamicArgumentsExpression)dynamicExpression;
+            if (expression.ArgumentRefKinds.IsDefault)
```

Comment:
Please extract a local for `expression.ArgumentRefKinds`. And consider using `dynamicExpression.ArgumentRefKinds` directly.

Reply:
Done